### ISSUE TITLE
--no-deps doesn't ignore path based deps

### ISSUE THREAD

**Issue by gluax** [rust-lang/rust-clippy#12658]
### Summary

I'm in a workspace for a project. And we have to clone one dependency and have our workspace.dependencies point to the path for that dep. However, warnings for that dep will still be printed:

![image](https://github.com/rust-lang/rust-clippy/assets/16431709/2d1d80ba-8935-4660-81c1-b32bdd55f46d)

### Reproducer

I expected no clippy warnings for a local dep.

Also, note that these warnings don't trigger the expected deny warning failure. It seems weird to me to see these warnings if they don't trigger a failure.

### Version

```text
rustc 1.77.1 (7cf61ebde 2024-03-27)
binary: rustc
commit-hash: 7cf61ebde7b22796c69757901dd346d0fe70bd97
commit-date: 2024-03-27
host: x86_64-unknown-linux-gnu
release: 1.77.1
LLVM version: 17.0.6
```

**Comment by Alexendoo** [comment]
`--no-deps` is applying here, it's causing `snarkvm` to be compiled by `rustc` instead of `clippy-driver`. The `dead_code` lint is from the rust compiler rather than Clippy

The `-D warnings` only gets applied when Clippy is ran on a package
